petab/v2/core.py

```diff
@@ -1138,8 +1138,13 @@ def __init__(
         self.mapping_tables = mapping_tables or [MappingTable()]
         self.parameter_tables = parameter_tables or [ParameterTable()]
 
+    def __repr__(self):
+        return f"<{self.__class__.__name__} id={self.id!r}>"
+
     def __str__(self):
-        model = f"with model ({self.model})" if self.model else "without model"
+        pid = repr(self.id) if self.id else "without ID"
```

Comment:
What is the benefit of `repr` here? `self.id` looks like it's either `str` or `None`, so `repr` and `!r` don't change anything?

Reply:
> What is the benefit of `repr` here? `self.id` looks like it's either `str` or `None`, so `repr` and `!r` don't change anything?

As compared to what? Not using `repr` here? The difference will be the quotes.

Reply:
Why not `pid = self.id or "without ID"` since `pid` is used as `f"...{pid}..."` later? Or what do you mean by the difference will be the quotes?

Reply:
> Why not `pid = self.id or "without ID"` since `pid` is used as `f"...{pid}..."` later? Or what do you mean by the difference will be the quotes?

Currently, the model ID is shown in quotation marks if it is not None. With this change, it would be shown unquoted. When additionally using `!r` below, the model ID would be quoted, but so will be the "without ID". (`print("foo", repr("foo"))` -> `foo 'foo'`)